Added hint prop, few styles, and made Widget more abstract.

sarmadka opened the pull request:
* Added the `title` property to `Widget`.
* Added `textOverflow` style.
* Added `whiteSpace` style.
* Added `Length.inherit`.
* Added `Length4.inherit`.
* Added missing Arabic localizations.
* Refactored the Widget class to make its properties and
  events abstract so they can be implemented by both basic
  widgets as well as components. This eliminates the
  disconnect in behavior between components and basic
  widgets. So now using those basic events and properties
  on components will automatically be applied to the root
  widget of that component.

sarmadka committed:
Added title property and few styles.

* Added the `title` property to `Widget`.
* Added `textOverflow` style.
* Added `whiteSpace` style.
* Added `Length.inherit`.
* Added `Length4.inherit`.
* Added missing Arabic localizations.

SlemanAlrstum231 commented on Doc/widgets.en.md:
```diff
@@ -17,6 +17,7 @@ The base class for all widgets.
 #### Properties
 
 * `className`: `String`. Class name of the widget, used for applying styles by class name.
+* `title`: `String`. A string to be displayed as a tooltip when the cursor hovers over the element.
```
اممم ما رايك اخي الكريم ان نغير اسمها لتلميح بدل عنوان hint

sarmadka replied:
دعني أفكر بالأمر قليلا. اخترت title لأنه اسم الخصلة المعتمدة في HTML فستكون مألوفة للمستخدم، لكن في الحقيقة كلمة title بعيدة عن المعنى الحقيقي لهذه الخصلة، وكلمة hint أقرب للمعنى الحقيقي.

sarmadka committed:
Renamed `title` property to `hint`.

sarmadka committed:
Made Widget operations abstract.

Refactored the Widget class to make its properties and
events abstract so they can be implemented by both basic
widgets as well as components. This eliminates the
disconnect in behavior between components and basic
widgets. So now using those basic events and properties
on components will automatically be applied to the root
widget of that component.

sarmadka committed:
Fixed a bug with Length's "inherit" value.

SlemanAlrstum231 commented on WebPlatform/Widgets/BasicWidget.alusus:
```diff
@@ -0,0 +1,273 @@
+            };
+
+            this.onTouchStartObj.onConnectionsChanged = closure (connectionCount: Int) {
+                handleSignalConnectionChanged[this.id, onTouchStart, "touchstart", {
```
هنا اخي الكريم من اجل اللمس ؟ الا يعتبر اللمس مثل الماوس اي ضغطة الماوس ؟

SlemanAlrstum231 commented on WebPlatform/Widgets/Widget.alusus:
```diff
@@ -3,245 +3,40 @@
     // Widget
     // The base class of all widgets.
     class Widget {
-        // Variables
```
ماذا اصبح عمله اخي الكريم بعد تعريف BASEWIDGETS
